src/aio/native_linux_aio_provider.cpp

```diff
@@ -68,14 +70,43 @@ error_code native_linux_aio_provider::flush(dsn_handle_t fh)
 error_code native_linux_aio_provider::write(const aio_context &aio_ctx,
                                             /*out*/ uint32_t *processed_bytes)
 {
-    ssize_t ret = pwrite(static_cast<int>((ssize_t)aio_ctx.file),
-                         aio_ctx.buffer,
-                         aio_ctx.buffer_size,
-                         aio_ctx.file_offset);
-    if (ret < 0) {
-        return ERR_FILE_OPERATION_FAILED;
+    uint32_t buffer_offset = 0;
+    uint32_t remaing_size = aio_ctx.buffer_size;
+    while (remaing_size > 0) {
```

Comment:
remove `remaining_size`. use buffer_offset < aio_ctx.buffer_size to see whether it is over or not